dts/xtensa/intel/intel_adsp_ace30_ptl.dtsi

```diff
@@ -205,6 +205,7 @@
 			#size-cells = <0>;
 			reg = <0x00028100 0x1000
 				   0x00079C00 0x200>;
+			i2svss = <0x00028C00 0x1000>;
```

Comment:
In Zephyr, everything Device Tree is performed at compile-time so I really wondered how it could work without this line. I think the answer is: because the initialization code should look more like this below.

I can't possibly imagine a configuration where it is correct to let the compiler initialize these values to zero by default.

I'm sharing this below because the same problem and bug is likely to repeat itself; it might have already elsewhere.

Any DT expert around and if yes any insight to share?

```diff
diff --git a/drivers/dai/intel/ssp/ssp.h b/drivers/dai/intel/ssp/ssp.h
index 560b5d3df5f6..bca7e1b51810 100644
--- a/drivers/dai/intel/ssp/ssp.h
+++ b/drivers/dai/intel/ssp/ssp.h
@@ -108,6 +108,10 @@ struct dai_intel_ssp_plat_fifo_data {
        uint32_t handshake;
 };

+#if defined(CONFIG_SOC_INTEL_ACE20_LNL) || defined(CONFIG_SOC_INTEL_ACE30_PTL)
+#define INTEL_SSP_EXTRA_BASES 1
+#endif
+
 struct dai_intel_ssp_plat_data {
        uint32_t ssp_index;
        int acquire_count;
@@ -116,7 +120,7 @@ struct dai_intel_ssp_plat_data {
        uint32_t base;
        uint32_t ip_base;
        uint32_t shim_base;
-#if defined(CONFIG_SOC_INTEL_ACE20_LNL) || defined(CONFIG_SOC_INTEL_ACE30_PTL)
+#if INTEL_SSP_EXTRA_BASES
        uint32_t hdamlssp_base;
        uint32_t i2svss_base;
 #endif
--- a/drivers/dai/intel/ssp/ssp.c
+++ b/drivers/dai/intel/ssp/ssp.c
@@ -69,7 +69,7 @@ static struct dai_intel_ssp_mn ssp_mn_divider = {
                .shim_base = DT_REG_ADDR_BY_IDX(DT_NODELABEL(shim), 0),         \
                IF_ENABLED(DT_NODE_EXISTS(DT_NODELABEL(hdamlssp)),              \
                        (.hdamlssp_base = DT_REG_ADDR(DT_NODELABEL(hdamlssp)),))\
-               IF_ENABLED(DT_PROP_HAS_IDX(node_id, i2svss, 0),                 \
+               IF_ENABLED(INTEL_SSP_EXTRA_BASES,                       \
                        (.i2svss_base = DT_PROP_BY_IDX(node_id, i2svss, 0),))   \
                .irq = DT_NUM_IRQS(node_id),                            \
                .irq_name = irq_name_level5_z,                          \
```